Panic when telegraf reloads on watch

### Relevant telegraf.conf

```toml
[[inputs.mem]]

[[outputs.http]]
```

### Logs from Telegraf

```text
2025-07-22T05:20:41Z I! Reloading Telegraf config
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x10b194504]

goroutine 11 [running]:
time.(*Timer).Stop(...)
        /usr/local/go/src/time/sleep.go:114
main.(*Telegraf).watchLocalConfig(0x140012b4f20, {0x10eedd550, 0x140014e51d0}, 0x14001271b90, {0x16bb4b42e, 0x9})
        /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:278 +0x734
created by main.(*Telegraf).reloadLoop in goroutine 1
        /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:169 +0x6e0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x10b194504]

goroutine 8 [running]:
time.(*Timer).Stop(...)
        /usr/local/go/src/time/sleep.go:114
main.(*Telegraf).watchLocalConfig(0x140012b4f20, {0x10eedd550, 0x140014e51d0}, 0x14001271b90, {0x140014ac3f0, 0x14})
        /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:278 +0x734
created by main.(*Telegraf).reloadLoop in goroutine 1
        /go/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:162 +0x570
```

### System info

Telegraf 1.35.2

### Docker

_No response_

### Steps to reproduce

1. Run telegraf with `--watch-config=notify`
2. Modify an existing config file loaded.
3. Observe that Telegraf crashes.


### Expected behavior

Should reload gracefully.

### Actual behavior

Crashes due to panic

### Additional info

Related to #17048

--- a/cmd/telegraf/telegraf.go
+++ b/cmd/telegraf/telegraf.go
@@ -327,7 +327,9 @@ func (t *Telegraf) watchLocalConfig(ctx context.Context, signals chan os.Signal,
 			}
 
 		case <-mytomb.Dying():
-			reloadTimer.Stop()
+			if reloadTimer != nil {
+				reloadTimer.Stop()
+			}
 			log.Printf("I! Config watcher %q ended\n", fConfig)
 			return
 		}

